src/components/subscription-plans/blocks/subscribers-table.tsx

```diff
@@ -0,0 +1,208 @@
+"use client";
+
+import { ShortAddress } from "@/components/short-address";
+import { Card, CardContent } from "@/components/ui/card";
+import {
+  Select,
+  SelectContent,
+  SelectItem,
+  SelectTrigger,
+  SelectValue,
+} from "@/components/ui/select";
+import {
+  Table,
+  TableBody,
+  TableCell,
+  TableHeader,
+  TableRow,
+} from "@/components/ui/table";
+import { CompletedPayments } from "@/components/view-recurring-payments/blocks/completed-payments";
+import { FrequencyBadge } from "@/components/view-recurring-payments/blocks/frequency-badge";
+import { formatCurrencyLabel } from "@/lib/constants/currencies";
+import type { SubscriptionWithDetails } from "@/lib/types";
+import type { SubscriptionPlan } from "@/server/db/schema";
+import { api } from "@/trpc/react";
+import { addDays, format } from "date-fns";
+import { CreditCard, DollarSign, Filter } from "lucide-react";
+import { useState } from "react";
+import { EmptyState } from "../../dashboard/blocks/empty-state";
+import { StatCard } from "../../dashboard/blocks/stat-card";
+import { TableHeadCell } from "../../dashboard/blocks/table-head-cell";
+
+interface SubscribersTableProps {
+  initialSubscribers: SubscriptionWithDetails[];
+  subscriptionPlans: SubscriptionPlan[];
+}
+
+const ACTIVE_STATUSES = ["pending", "active"];
+
+const SubscriberTableColumns = () => (
+  <TableRow className="hover:bg-transparent border-none">
+    <TableHeadCell>Start Date</TableHeadCell>
+    <TableHeadCell>Plan Name</TableHeadCell>
+    <TableHeadCell>Trial Info</TableHeadCell>
+    <TableHeadCell>Frequency</TableHeadCell>
+    <TableHeadCell>Currency</TableHeadCell>
+    <TableHeadCell>Subscriber</TableHeadCell>
+    <TableHeadCell>Payment History</TableHeadCell>
+  </TableRow>
+);
+
+const SubscriberRow = ({
+  subscription,
+}: { subscription: SubscriptionWithDetails }) => {
+  const getTrialEndDate = () => {
+    if (!subscription.subscription?.trialDays) return "No trial";
+    if (!subscription.createdAt) return "No trial";
+
+    const trialEndDate = addDays(
+      new Date(subscription.createdAt),
+      subscription.subscription.trialDays,
+    );
+    return format(trialEndDate, "do MMM yyyy");
+  };
+
+  return (
+    <TableRow className="hover:bg-zinc-50/50">
+      <TableCell>
+        {subscription.createdAt
+          ? format(new Date(subscription.createdAt), "do MMM yyyy")
+          : "N/A"}
+      </TableCell>
+      <TableCell className="font-medium">
+        {subscription.subscription?.label || "Unnamed Plan"}
+      </TableCell>
+      <TableCell>
+        {subscription.subscription?.trialDays
+          ? `${subscription.subscription.trialDays} days (ends ${getTrialEndDate()})`
+          : "No trial"}
```

Comment:
Redundant Trial Check: The component checks subscription.subscription?.trialDays twice - once here and once inside getTrialEndDate(). Since getTrialEndDate() already handles the no-trial case, you can simplify this to just call the function:
```suggestion
        {getTrialEndDate()}
```